### PR TITLE
Update AWS deployment resources for version 1.2.0

### LINKED ISSUE #191: Update cloud-front / ELB configuration to forward request headers to registry-api
## 💡 Description

Find what is the configuration equivalent to what has been tested in nginx in https://github.com/NASA-PDS/registry/blob/b056cf7008c6b04efcca9c76ac61921744df4a9f/docker/docker-compose.yml#L238

See repository: https://github.com/NASA-PDS/proxycrypt/blob/main/etc/default.conf.template

### LINKED ISSUE #330: Make the registry-api 1.2.0 deployment work on gamma
## 💡 Description

- [x] This require to upgrate the AWS configuration for the new version (application.properties, Dockerfile.aws)
- [x] Update the cloudfront configuration to send necessary X-Forwarded-* headers

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary
- The Dockerfile.aws has been updated to use jdk17
- The application.properties.aws has been updated to match updates of the main application.properties
- Not In The PR: The cloudfront function used to add headers to the incoming request has been updated and tested.

The deployment has been tested on gamma with the newly generated docker image, see:
https://pds.nasa.gov/api/search-en-gamma/1/

## ⚙️ Test Data and/or Report
One of the following should be included here:
* Reference to regression test included in code (preferred wherever reasonable)
* Attach test data here + outputs of tests

## ♻️ Related Issues

fixes https://github.com/NASA-PDS/registry-api/issues/330
fixes https://github.com/NASA-PDS/registry/issues/191

### CODE REVIEW

File: .github/workflows/stable-cicd.yaml
Comment (alexdunnjpl) [inline .github/workflows/stable-cicd.yaml:1]: request: revert this change - the post-diff approach is deprecated

**Review by alexdunnjpl — CHANGES_REQUESTED** [review]
Looks good besides the workflow change

File: .github/workflows/stable-cicd.yaml
Comment (tloubrieu-jpl) [reply .github/workflows/stable-cicd.yaml:1]: ok, will do

**Review by alexdunnjpl — APPROVED** [review]
